include/nbl/builtin/hlsl/functional.hlsl

```diff
@@ -81,48 +93,47 @@ struct mul
 };
 
 template<typename T>
-struct comparator_lt_t
+struct greater
 {
     bool operator()(const T lhs, const T rhs)
     {
-        return lhs<rhs;
+        return lhs > rhs;
     }
 };
 
 template<typename T>
-struct comparator_gt_t
+struct less
 {
     bool operator()(const T lhs, const T rhs)
     {
-        return lhs>rhs;
+        return lhs < rhs;
     }
 };
 
 template<typename T>
-struct comparator_lte_t
+struct greater_equal
 {
     bool operator()(const T lhs, const T rhs)
     {
-        return lhs<=rhs;
+        return lhs >= rhs;
     }
 };
 
 template<typename T>
-struct comparator_gte_t
+struct less_equal
 {
     bool operator()(const T lhs, const T rhs)
     {
-        return lhs>=rhs;
+        return lhs <= rhs;
     }
 };
 
 template<typename T>
-struct min
+struct minimum
 {
```

Comment:
since these don't exist in `std`/C++ they should be declared outside of the `#ifndef`